<launch>
    <arg name="show" default="false" />
    <arg name="output" default="screen" />
    <arg name="manager"     default="swarm_manager"/>
    <arg name="send_img"     default="false"/>
    <arg name="send_whole_img_desc"     default="false"/>
    <arg name="self_id"     default="1"/>
    <arg name="camera_config_path" default="/home/dji/SwarmConfig/fisheye_ptgrey_n3/front.yaml" />
    <arg name="vins_config_path" default="/home/dji/SwarmConfig/fisheye_ptgrey_n3/fisheye_cuda.yaml" />
    <node pkg="nodelet" type="nodelet" name="swarm_loop" 
        args="load D2Frontend/D2FrontendNode /$(arg manager) --no-bond" output="screen" >
        <rosparam>
            lcm_uri: udpm://224.0.0.251:7667?ttl=1
            init_loop_min_feature_num: 30
            min_loop_feature_num: 30
            jpg_quality: 75
            
            query_thres: 0.3
            init_query_thres: 0.2

            min_movement_keyframe: 0.1
            enable_pub_remote_img: false
            enable_sub_remote_img: false
            max_freq: 0.3
            accept_min_3d_pts: 30
            match_index_dist: 5
            recv_msg_duration: 0.5
            superpoint_thres: 0.02
            triangle_thres: 0.012
            superpoint_max_num: 200
            #triangle_thres: 0.008
            min_direction_loop: 3
            
            detector_match_thres: 0.7
            output_path: /root/output/
            send_all_features: false
            lower_cam_as_main: false
            min_match_per_dir: 5
            camera_configuration: 1
            output_raw_superpoint_desc: false
            loop_cov_pos: 1e-2 #Indoor, 600px width
            odometry_consistency_threshold: 2.8
            pos_covariance_per_meter: 0.0009
            yaw_covariance_per_meter: 0.00005
            width: 400
            height: 208
        </rosparam>
    <param name="pca_comp_path" value="$(find swarm_loop)/models/components_.csv" type="string" />
    <param name="pca_mean_path" value="$(find swarm_loop)/models/mean_.csv" type="string" />
    <param name="superpoint_model_path" value="$(find swarm_loop)/models/superpoint_v1_tx2_fp16.trt" type="string" />
    <param name="netvlad_model_path" value="$(find swarm_loop)/models/mobilenetvlad_208x400_tx2_fp16.trt" type="string" />
    <param name="show" value="$(arg show)" type="bool" />
    <param name="send_img" value="$(arg send_img)" type="bool" />
    <param name="send_whole_img_desc" value="$(arg send_whole_img_desc)" type="bool" />
    <param name="self_id" value="$(arg self_id)" type="int" />
    <param name="camera_config_path" value="$(arg camera_config_path)" type="string" />
    <param name="vins_config_path" value="$(arg vins_config_path)" type="string" />

    </node>
</launch>
